src/mtype.d

```diff
@@ -7963,13 +7948,14 @@ public:
                 return oe;
             return new DotExp(e.loc, e, oe);
         }
+
         Declaration d = s.isDeclaration();
-        debug
+        if (!d)
         {
-            if (!d)
-                printf("d = %s '%s'\n", s.kind(), s.toChars());
+            e.error("%s.%s is not a declaration", e.toChars(), ident.toChars());
```

Comment:
The reason for the `assert(d)` was that this should never happen. Replacing it with `error` means it does happen - but there is no test case showing it can be.